### ISSUE TITLE
[Bug] Unable to find destination _error_pointing_to_this_heading while generating PDF outline! This likely happened because an anchor link contained an umlaut 

### ISSUE THREAD

**Issue by mariotoffia** [Mogztter/asciidoctor-web-pdf#340]
Thanks for this *very* nice pdf generator!!

I think I've stumbeled upon a bug. When I have the below file, I get the following error while trying to generate PDF.

**- "Unable to layout item:  JSHandle@node
Unable to find destination _error_pointing_to_this_heading while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254)."**

the generation is done via this command
```bash
node_modules/asciidoctor-pdf/bin/asciidoctor-pdf \ 
        -r asciidoctor-kroki \
        -r asciidoctor-highlight.js \
        -a revnumber="`git describe --tags`" \
        -a revdate="`git log -1 --format=%cd`" \
        -a revremark="`git log -1 --pretty=%B`" \
        -a kroki-server-url="http://localhost:8000" \
        go-core-generated.adoc \
        -o ../_dist/_output/docs/go-core-api.pdf
```

file: go-core-generated.adoc
```adoc
:toc:
:toc-title: Error output to PDF
:toclevels: 2
:icons: font
:doctype: book

= Sample

== Variables

=== abbreviationToIANA
[source, go]
----
var abbreviationToIANA = map[string]string{
    "Egypt Standard Time":             "Africa/Cairo",
    "Morocco Standard Time":           "Africa/Casablanca",
    "South Africa Standard Time":      "Africa/Johannesburg",
    "Sudan Standard Time":             "Africa/Khartoum",
    "W. Central Africa Standard Time": "Africa/Lagos",
    "E. Africa Standard Time":         "Africa/Nairobi",
    "Sao Tome Standard Time":          "Africa/Sao_Tome",
    "Libya Standard Time":             "Africa/Tripoli",
    "Namibia Standard Time":           "Africa/Windhoek",
    "Aleutian Standard Time":          "America/Adak",
    "Alaskan Standard Time":           "America/Anchorage",
    "Tocantins Standard Time":         "America/Araguaina",
    "Paraguay Standard Time":          "America/Asuncion",
    "Bahia Standard Time":             "America/Bahia",
    "SA Pacific Standard Time":        "America/Bogota",
    "Argentina Standard Time":         "America/Buenos_Aires",
    "Eastern Standard Time (Mexico)":  "America/Cancun",
    "Venezuela Standard Time":         "America/Caracas",
    "SA Eastern Standard Time":        "America/Cayenne",
    "Central Standard Time":           "America/Chicago",
    "Mountain Standard Time (Mexico)": "America/Chihuahua",
    "Central Brazilian Standard Time": "America/Cuiaba",
    "Mountain Standard Time":          "America/Denver",
    "Greenland Standard Time":         "America/Godthab",
    "Turks And Caicos Standard Time":  "America/Grand_Turk",
    "Central America Standard Time":   "America/Guatemala",
    "Atlantic Standard Time":          "America/Halifax",
    "Cuba Standard Time":              "America/Havana",
    "US Eastern Standard Time":        "America/Indianapolis",
    "SA Western Standard Time":        "America/La_Paz",
    "Pacific Standard Time":           "America/Los_Angeles",
    "Central Standard Time (Mexico)":  "America/Mexico_City",
    "Saint Pierre Standard Time":      "America/Miquelon",
    "Montevideo Standard Time":        "America/Montevideo",
    "Eastern Standard Time":           "America/New_York",
    "US Mountain Standard Time":       "America/Phoenix",
    "Haiti Standard Time":             "America/Port-au-Prince",
    "Magallanes Standard Time":        "America/Punta_Arenas",
    "Canada Central Standard Time":    "America/Regina",
    "Pacific SA Standard Time":        "America/Santiago",
    "E. South America Standard Time":  "America/Sao_Paulo",
    "Newfoundland Standard Time":      "America/St_Johns",
    "Pacific Standard Time (Mexico)":  "America/Tijuana",
    "Central Asia Standard Time":      "Asia/Almaty",
    "Jordan Standard Time":            "Asia/Amman",
    "Arabic Standard Time":            "Asia/Baghdad",
    "Azerbaijan Standard Time":        "Asia/Baku",
    "SE Asia Standard Time":           "Asia/Bangkok",
    "Altai Standard Time":             "Asia/Barnaul",
    "Middle East Standard Time":       "Asia/Beirut",
    "India Standard Time":             "Asia/Calcutta",
    "Transbaikal Standard Time":       "Asia/Chita",
    "Sri Lanka Standard Time":         "Asia/Colombo",
    "Syria Standard Time":             "Asia/Damascus",
    "Bangladesh Standard Time":        "Asia/Dhaka",
    "Arabian Standard Time":           "Asia/Dubai",
    "West Bank Standard Time":         "Asia/Hebron",
    "W. Mongolia Standard Time":       "Asia/Hovd",
    "North Asia East Standard Time":   "Asia/Irkutsk",
    "Israel Standard Time":            "Asia/Jerusalem",
    "Afghanistan Standard Time":       "Asia/Kabul",
    "Russia Time Zone 11":             "Asia/Kamchatka",
    "Pakistan Standard Time":          "Asia/Karachi",
    "Nepal Standard Time":             "Asia/Katmandu",
    "North Asia Standard Time":        "Asia/Krasnoyarsk",
    "Magadan Standard Time":           "Asia/Magadan",
    "N. Central Asia Standard Time":   "Asia/Novosibirsk",
    "Omsk Standard Time":              "Asia/Omsk",
    "North Korea Standard Time":       "Asia/Pyongyang",
    "Qyzylorda Standard Time":         "Asia/Qyzylorda",
    "Myanmar Standard Time":           "Asia/Rangoon",
    "Arab Standard Time":              "Asia/Riyadh",
    "Sakhalin Standard Time":          "Asia/Sakhalin",
    "Korea Standard Time":             "Asia/Seoul",
    "China Standard Time":             "Asia/Shanghai",
    "Singapore Standard Time":         "Asia/Singapore",
    "Russia Time Zone 10":             "Asia/Srednekolymsk",
    "Taipei Standard Time":            "Asia/Taipei",
    "West Asia Standard Time":         "Asia/Tashkent",
    "Georgian Standard Time":          "Asia/Tbilisi",
    "Iran Standard Time":              "Asia/Tehran",
    "Tokyo Standard Time":             "Asia/Tokyo",
    "Tomsk Standard Time":             "Asia/Tomsk",
    "Ulaanbaatar Standard Time":       "Asia/Ulaanbaatar",
    "Vladivostok Standard Time":       "Asia/Vladivostok",
    "Yakutsk Standard Time":           "Asia/Yakutsk",
    "Ekaterinburg Standard Time":      "Asia/Yekaterinburg",
    "Caucasus Standard Time":          "Asia/Yerevan",
    "Azores Standard Time":            "Atlantic/Azores",
    "Cape Verde Standard Time":        "Atlantic/Cape_Verde",
    "Greenwich Standard Time":         "Atlantic/Reykjavik",
    "Cen. Australia Standard Time":    "Australia/Adelaide",
    "E. Australia Standard Time":      "Australia/Brisbane",
    "AUS Central Standard Time":       "Australia/Darwin",
    "Aus Central W. Standard Time":    "Australia/Eucla",
    "Tasmania Standard Time":          "Australia/Hobart",
    "Lord Howe Standard Time":         "Australia/Lord_Howe",
    "W. Australia Standard Time":      "Australia/Perth",
    "AUS Eastern Standard Time":       "Australia/Sydney",
    "UTC":                             "Etc/GMT",
    "UTC-11":                          "Etc/GMT+11",
    "Dateline Standard Time":          "Etc/GMT+12",
    "UTC-02":                          "Etc/GMT+2",
    "UTC-08":                          "Etc/GMT+8",
    "UTC-09":                          "Etc/GMT+9",
    "UTC+12":                          "Etc/GMT-12",
    "UTC+13":                          "Etc/GMT-13",
    "Astrakhan Standard Time":         "Europe/Astrakhan",
    "W. Europe Standard Time":         "Europe/Berlin",
    "GTB Standard Time":               "Europe/Bucharest",
    "Central Europe Standard Time":    "Europe/Budapest",
    "E. Europe Standard Time":         "Europe/Chisinau",
    "Turkey Standard Time":            "Europe/Istanbul",
    "Kaliningrad Standard Time":       "Europe/Kaliningrad",
    "FLE Standard Time":               "Europe/Kiev",
    "GMT Standard Time":               "Europe/London",
    "Belarus Standard Time":           "Europe/Minsk",
    "Russian Standard Time":           "Europe/Moscow",
    "Romance Standard Time":           "Europe/Paris",
    "Russia Time Zone 3":              "Europe/Samara",
    "Saratov Standard Time":           "Europe/Saratov",
    "Volgograd Standard Time":         "Europe/Volgograd",
    "Central European Standard Time":  "Europe/Warsaw",
    "Mauritius Standard Time":         "Indian/Mauritius",
    "Samoa Standard Time":             "Pacific/Apia",
    "New Zealand Standard Time":       "Pacific/Auckland",
    "Bougainville Standard Time":      "Pacific/Bougainville",
    "Chatham Islands Standard Time":   "Pacific/Chatham",
    "Easter Island Standard Time":     "Pacific/Easter",
    "Fiji Standard Time":              "Pacific/Fiji",
    "Central Pacific Standard Time":   "Pacific/Guadalcanal",
    "Hawaiian Standard Time":          "Pacific/Honolulu",
    "Line Islands Standard Time":      "Pacific/Kiritimati",
    "Marquesas Standard Time":         "Pacific/Marquesas",
    "Norfolk Standard Time":           "Pacific/Norfolk",
    "West Pacific Standard Time":      "Pacific/Port_Moresby",
    "Tonga Standard Time":             "Pacific/Tongatapu",
}
----


== Error pointing to this heading
```

**Comment by Mogztter** [comment]
Hello @mariotoffia and thanks for your kind words.
This is actually a duplicate of https://github.com/Mogztter/asciidoctor-web-pdf/issues/336#issuecomment-709350834

I found a workaround using a custom stylesheet:

```css
.listingblock {
  break-inside: auto;
}
```

Maybe we could add this rule in the default stylesheet (until there's a fix in Paged.js) but I need to do some testing to make sure that it won't have side effects.

**Comment by boydkelly** [comment]
Hi,
I'm getting this error as well...  Unable to find destination _menteur_inegalé while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254).

And I assume that e accute is also causing a problem, not just umlaut.  However visiting the chromium bug link above suggests the bug was fixed on August 27th 2020.   I have a lot of not only French but also Latin extended characters.   Should this maybe work with a more recent chrome?   Or is there any other work around?  Thanks!

**Comment by Mogztter** [comment]
Hi @boydkelly,

This error can be misleading because it will also warn when the document has been truncated or more precisely when Paged.js was unable to fit all the blocks/content into pages.
For instance, if you have a section with the title "Menteur inegalé" and this section is missing from the output (for whatever reasons) then you will get the warning message "Unable to find destination _menteur_inegalé..."

Could you please check your PDF and tell us if something is missing? In addition, could you please share your document or a sample document that reproduces this issue?

Thanks!

**Comment by h-a-t** [comment]
Oh hai o/
I can confirm that something is going on when using other languages in titles.
Using orly example :

```
= Hoping This Works
:doctype: book

== Guessing en français se dit deviner

The first step is guessing.

```
Compiling it in windows gives the following warning and build the pdf: 
```
PS D:\code\bin\asciidoctor-web-pdf\examples\document> ..\..\asciidoctor-web-pdf.exe .\orly.adoc -a stylesheet="+orly.css"
Unable to find destination _guessing_en_français_se_dit_deviner while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254).
```
Hope that helps o/

**Comment by Mogztter** [comment]
@h-a-t That definitely helped!
I was under the impression that this bug was only occurring when a section was absent from the output but apparently I was wrong.

I did some tests and I found that encoding section IDs seems to be enough. Take a look at: https://github.com/Mogztter/asciidoctor-web-pdf/blob/issue-340-encode-section-id/test/fixtures/section-ids.pdf (you will need to download the file)

**Comment by h-a-t** [comment]
@Mogztter LGTM! Merci Guillaume !
Taking the opportunity to thank you and the whole team for giving us the opportunity to mix JS with a bunch of Asciidoc magic in order to make our personal/professional writing workflow more streamlined \o/
Cheers

**Comment by boydkelly** [comment]
> @h-a-t That definitely helped!
> I was under the impression that this bug was only occurring when a section was absent from the output but apparently I was wrong.
> 
> I did some tests and I found that encoding section IDs seems to be enough. Take a look at: https://github.com/Mogztter/asciidoctor-web-pdf/blob/issue-340-encode-section-id/test/fixtures/section-ids.pdf (you will need to download the file)

Je viens de voir ceci!   Merci beaucoup!

**Comment by dgslos** [comment]
I'm still getting this error. Downloaded the v1.0.0-alpha.14 release. Tried both linux and windows versions

Error:
Unable to layout item:  JSHandle@node
Unable to find destination _overview_size_of_the_subcollection_mostly_by_year_if_database_in_access while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254).
Unable to find destination _additional_questions while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254).
Unable to find destination _possible_improvements while generating PDF outline! This likely happened because an anchor link contained an umlaut (https://bugs.chromium.org/p/chromium/issues/detail?id=985254).

**Comment by Mogztter** [comment]
@dgslos This is a generic error message when a section title is missing. In your case, I don't think it's related to an anchor link containing an umlaut. Could you please open a new issue with your document ?

**Comment by dgslos** [comment]
@Mogztter, the .adoc is a working document and meant to be open in the public. Can I send it to you by mail?

**Comment by Mogztter** [comment]
@dgslos Sure, not sure if my personal email address is public but you can use my company contact email address: https://www.yuzutech.fr/#contact

**Comment by dgslos** [comment]
Sended the adoc file to that email :)

**Comment by Mogztter** [comment]
Thanks and received, I will take a look when I have time.

**Comment by binaryflare** [comment]
I'm very glad I came across this bug.  I ran into this issue and the problem was probably this: "This error can be misleading because it will also warn when the document has been truncated or more precisely when Paged.js was unable to fit all the blocks/content into pages."    It was a long horizontal description list (a list of acryonyms in this case) that probably should span multiple pages.  At the moment, to avoid the issue, I just changed the separator to a dash rather than using "::" and thus getting separate paragraphs.  I haven't tried the custom stylesheet solution mentioned here yet.
